### ISSUE TITLE
Add support for trusted-types and require-trusted-types

### ISSUE THREAD

**Issue by kingthorin** [HtmlUnit/htmlunit-csp#4]
It would be nice to have handling for: the ~~[`trusted-types`](https://w3c.github.io/webappsec-trusted-types/dist/spec/#trusted-types-csp-directive)~~ [`trusted-types`](https://www.w3.org/TR/trusted-types/) and ~~[`require-trusted-types-for`](https://w3c.github.io/webappsec-trusted-types/dist/spec/#require-trusted-types-for-csp-directive)~~ [require-trusted-types-for](https://www.w3.org/TR/trusted-types/#require-trusted-types-for-csp-directive) directives.

This advice was provided pre-fork:

-----

The process will be basically

- define new ~~[Value](https://github.com/shapesecurity/salvation/tree/d867e194e63eb48a7c82bec31fe1edb4ad407b2c/src/main/java/com/shapesecurity/salvation2/Values)~~ [Value](https://github.com/HtmlUnit/htmlunit-csp/tree/main/src/main/java/org/htmlunit/csp/value) kinds for new directive values which are more complex than just a boolean. At a glance, you probably want a special Value kind for `tt-policy-name`, but the others look like they can be handled as booleans on the directive.
- define new ~~[Directives](https://github.com/shapesecurity/salvation/tree/d867e194e63eb48a7c82bec31fe1edb4ad407b2c/src/main/java/com/shapesecurity/salvation2/Directives)~~ [Directive](https://github.com/HtmlUnit/htmlunit-csp/tree/main/src/main/java/org/htmlunit/csp/directive) kind for both new directives (~~[`trusted-types`](https://w3c.github.io/webappsec-trusted-types/dist/spec/#trusted-types-csp-directive)~~ [`trusted-types`](https://www.w3.org/TR/trusted-types/)) and (~~[`require-trusted-types-for`](https://w3c.github.io/webappsec-trusted-types/dist/spec/#require-trusted-types-for-csp-directive)~~ [require-trusted-types-for](https://www.w3.org/TR/trusted-types/#require-trusted-types-for-csp-directive))
- add helpers to allow reading ~~and writing~~ their attributes, and ensure that updating keeps their internal `values` field up to date (by using the `addValue` and `removeValueIgnoringCase` helpers, mainly)
- define new ~~[Value](https://github.com/shapesecurity/salvation/tree/d867e194e63eb48a7c82bec31fe1edb4ad407b2c/src/main/java/com/shapesecurity/salvation2/Values)~~ [Value](https://github.com/HtmlUnit/htmlunit-csp/tree/main/src/main/java/org/htmlunit/csp/value) kinds for new directive values which are more complex than just a boolean. At a glance, you probably want a special Value kind for `tt-policy-name`, but the others look like they can be handled as booleans on the directive.
- define new ~~[Directives](https://github.com/shapesecurity/salvation/tree/d867e194e63eb48a7c82bec31fe1edb4ad407b2c/src/main/java/com/shapesecurity/salvation2/Directives)~~ [Directive](https://github.com/HtmlUnit/htmlunit-csp/tree/main/src/main/java/org/htmlunit/csp/directive) kind for both new directives (["trusted-types"](https://w3c.github.io/webappsec-trusted-types/dist/spec/#trusted-types-csp-directive) and (["require-trusted-types-for"](https://w3c.github.io/webappsec-trusted-types/dist/spec/#require-trusted-types-for-csp-directive))
- add helpers to allow reading ~~and writing~~ their attributes, and ensure that updating keeps their internal `values` field up to date (by using the `addValue` and `removeValueIgnoringCase` helpers, mainly)
- update the parser to handle them
- add tests

* optionally add helpers for queries like ~~["Should Trusted Type policy creation be blocked by Content Security Policy?"](https://w3c.github.io/webappsec-trusted-types/dist/spec/#should-block-create-policy)~~ [`Should Trusted Type policy creation be blocked by Content Security Policy?`](https://www.w3.org/TR/trusted-types/#should-block-create-policy), though these aren't so important. Update the parser to handle them

The main challenges in my experience is figuring out what the spec is trying to say and whether that's actually what browsers do. For example, it says the syntax is defined by an [ABNF](https://datatracker.ietf.org/doc/html/rfc5234), which means keywords are case-insensitive ("ABNF strings are case insensitive"). So per spec, `'ALLow-DUPlicATES'` is a legal way to write `'allow-duplicates'`. Is that actually intentional? Is that what browsers do? Gotta check. Similarly, are policy names case-sensitive or not? That affects the representation ~~and especially the manipulation~~ of those Values.

**Comment by kingthorin** [comment]
FYI @zaproxy is now using this library :grinning:
https://github.com/zaproxy/zap-extensions/pull/5231
